src/OrbitBase/TaskGroupTest.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+#include <stddef.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/TaskGroup.h"
+#include "OrbitBase/UniqueResource.h"
+
+using orbit_base::TaskGroup;
+using orbit_base::ThreadPool;
+
+namespace {
+
+void ShutdownThreadPool(std::shared_ptr<ThreadPool> thread_pool) { thread_pool->ShutdownAndWait(); }
+
+[[nodiscard]] ThreadPool* GetTestThreadPool() {
+  constexpr size_t kNumThreads = 4;
+  constexpr absl::Duration kThreadTtl = absl::Milliseconds(5);
+  static orbit_base::unique_resource handle{
+      ThreadPool::Create(kNumThreads, kNumThreads, kThreadTtl), ShutdownThreadPool};
+  return handle.get().get();
+}
```

Comment:
You could potentially use `Executor` and an implementation of `MockExecutor` for your tests (see ExecutorTest.cpp for examples), but I don't feel strongly about that. At least it would avoid the need to handle a threadpool which is cumbersome as you obviously realized. :sweat_smile:

Reply:
Yeah the handle was annoying, will leave it as is for now. I was thinking that we might want to make sure "ShutdownAndWait()" is called in the ThreadPool destructor, not sure why it's not the case already...